src/Compilers/CSharp/Test/Emit3/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -4441,5 +4442,203 @@ record B(int I) : A(I);
             var diagnostics = await compWithAnalyzers.GetAnalyzerSemanticDiagnosticsAsync(model, filterSpan: null, CancellationToken.None);
             diagnostics.Verify(Diagnostic("ID0001", "B").WithLocation(1, 8));
         }
+
+        private sealed class OptionsOverrideDiagnosticAnalyzer(AnalyzerConfigOptionsProvider customOptions) : DiagnosticAnalyzer
```

Comment:
these tests validate that we see the different options through all the callbacks.